src/arrayMethodJoin.js

```diff
@@ -5,7 +5,28 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let resultString = '';
+    let sepInsert = separator;
+
+    if (separator === undefined) {
+      sepInsert = ',';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      let element = this[i];
+
+      if (element === undefined || element == null) {
+        element = '';
+      }
+
+      if (i !== this.length - 1) {
+        resultString += `${element}${sepInsert}`;
+      } else {
+        resultString += `${element}`;
+      }
```

Comment:
```suggestion
      if (i !== this.length - 1) {
        resultString += separator;
      }
```

You can avoid else statement if you change the logic a little